lightning/config/config.go

```diff
@@ -176,38 +180,92 @@ func NewConfig() *Config {
+		},
 	}
 }
 
 func LoadConfig(args []string) (*Config, error) {
 	cfg := NewConfig()
 
-	cfg.FlagSet = flag.NewFlagSet("lightning", flag.ContinueOnError)
-	fs := cfg.FlagSet
+	fs := flag.NewFlagSet("lightning", flag.ContinueOnError)
 
 	// if both `-c` and `-config` are specified, the last one in the command line will take effect.
 	// the default value is assigned immediately after the StringVar() call,
 	// so it is fine to not give any default value for `-c`, to keep the `-h` page clean.
 	fs.StringVar(&cfg.ConfigFile, "c", "", "(deprecated alias of -config)")
-	fs.StringVar(&cfg.ConfigFile, "config", "tidb-lightning.toml", "tidb-lightning configuration file")
-	fs.BoolVar(&cfg.DoCompact, "compact", false, "do manual compaction on the target cluster, run then exit")
-	fs.StringVar(&cfg.SwitchMode, "switch-mode", "", "switch tikv into import mode or normal mode, values can be ['import', 'normal'], run then exit")
-	fs.BoolVar(&cfg.printVersion, "V", false, "print version of lightning")
+	fs.StringVar(&cfg.ConfigFile, "config", "", "tidb-lightning configuration file")
+	printVersion := fs.Bool("V", false, "print version of lightning")
+
+	logLevel := fs.String("L", "", `log level: info, debug, warn, error, fatal (default "info")`)
+	logFilePath := fs.String("log-file", "", "log file path")
+	tidbHost := fs.String("tidb-host", "", "TiDB server host")
+	tidbPort := fs.Int("tidb-port", 0, "TiDB server port (default 4000)")
+	tidbUser := fs.String("tidb-user", "", "TiDB user name to connect")
+	tidbStatusPort := fs.Int("tidb-status", 0, "TiDB server status port (default 10080)")
+	pdAddr := fs.String("pd-urls", "", "PD endpoint address")
+	dataSrcPath := fs.String("d", "", "Directory of the dump to import")
+	importerAddr := fs.String("importer", "", "address (host:port) to connect to tikv-importer")
 
 	if err := fs.Parse(args); err != nil {
 		return nil, errors.Trace(err)
 	}
+	if *printVersion {
+		fmt.Println(common.GetRawInfo())
+		return nil, flag.ErrHelp
+	}
 
 	if err := cfg.Load(); err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	if *logLevel != "" {
+		cfg.App.LogConfig.Level = *logLevel
+	}
+	if *logFilePath != "" {
+		cfg.App.LogConfig.File = *logFilePath
+	}
+	if *tidbHost != "" {
+		cfg.TiDB.Host = *tidbHost
+	}
+	if *tidbPort != 0 {
+		cfg.TiDB.Port = *tidbPort
+	}
+	if *tidbStatusPort != 0 {
+		cfg.TiDB.StatusPort = *tidbStatusPort
+	}
+	if *tidbUser != "" {
+		cfg.TiDB.User = *tidbUser
+	}
+	if *pdAddr != "" {
+		cfg.TiDB.PdAddr = *pdAddr
+	}
+	if *dataSrcPath != "" {
+		cfg.Mydumper.SourceDir = *dataSrcPath
+	}
+	if *importerAddr != "" {
+		cfg.TikvImporter.Addr = *importerAddr
+	}
+
+	cfg.Adjust()
+
 	return cfg, nil
 }
 
 func (cfg *Config) Load() error {
-	if cfg.printVersion {
-		fmt.Println(common.GetRawInfo())
-		return flag.ErrHelp
+	// search the default config path if none is specified.
+	if cfg.ConfigFile == "" {
+		for _, path := range defaultConfigPaths {
+			if _, err := os.Stat(path); err == nil {
```

Comment:
not good design, if user doesn't specify config file path, we should use default configuration values, its more nature

Reply:
The behavior is inherited from 2.x where we default to read form `tidb-lightning.toml` if `-config` is not specified. We could break this behavior?